hazelcast/src/main/java/com/hazelcast/map/impl/OwnedEntryCostEstimator.java

```diff
@@ -17,17 +17,20 @@
 package com.hazelcast.map.impl;
 
 /**
- * Size estimator general contract.
+ * Owned entry cost estimator general contract.
  *
- * @param <T> the type of object which's size going to be estimated.
+ * @param <K> the type of key which's size going to be estimated.
+ * @param <V> the type of key which's size going to be estimated.
  */
-public interface SizeEstimator<T> {
+public interface OwnedEntryCostEstimator<K, V> {
```

Comment:
`owned` in the name can be confusing. Because size estimator can also calculate backup entry costs or near cached entry costs. Maybe naming it as `MemoryCostEstimator` fits better?